docs/src/main/asciidoc/websockets-next-reference.adoc

```diff
@@ -846,6 +846,40 @@ quarkus.http.auth.permission.http-upgrade.paths=/end
 quarkus.http.auth.permission.http-upgrade.policy=authenticated
 ----
 
+Security annotations used during authentication must be placed on an endpoint class as well, for the `SecurityIdentity` is created before the websocket connection is opened.
+
+.Select Bearer token authentication mechanism
+[source, java]
+----
+package io.quarkus.websockets.next.test.security;
+
+import io.quarkus.oidc.BearerTokenAuthentication;
+import io.quarkus.websockets.next.OnTextMessage;
+import io.quarkus.websockets.next.WebSocket;
+
+@BearerTokenAuthentication <1>
+@WebSocket(path = "/end")
+public class Endpoint {
+
+    @OnTextMessage
+    String echo(String message) {
+        return message;
+    }
+
+}
+----
+<1> Require that opening WebSocket handshake request is authenticated using Bearer token authentication.
+Find how to select other authentication mechanisms with annotations in the xref:security-authentication-mechanisms.adoc#use-annotations-for-path-based-auth[Authentication mechanisms in Quarkus] guide.
+
+[source,properties]
+----
+quarkus.http.auth.proactive=false <1>
+----
+<1> Ensures that Quarkus matches opening WebSocket handshake request with the `io.quarkus.oidc.BearerTokenAuthentication` annotation before authentication begins.
+
+Another example of a security annotation that must be placed on the class-level is the `io.quarkus.oidc.Tenant` annotation, which is used for resolving the OIDC tenant identifiers.
```

Comment:
Michal, if we do talk about `@Tenant` here, than your original code example showing it would've been best, but you agreed to remove it from the actual code. I'm sorry for being pedantic :-), but the OIDC multi-tenancy is not relevant in this doc, OIDC users who work with the multi-tenancy will find your update in the OIDC multi-tenancy doc of help. 
(I think the situation where we have WebSocket Next combined with more than one OIDC provider a very advanced and therefore rather rare case).
Proposing to drop this block

Reply:
np, let's drop it